# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: netwerk/protocol/http/TRRServiceChannel.cpp
# Commit: b3ffe3409712
# Full Hash: b3ffe3409712f31bcf61cf93aa09bf137e45f9bc
# Author: Kershaw Chang <kershaw@mozilla.com>
# Date: 2025-03-29 09:19:43
# Description:
#   Bug 1951554 - Avoid racing on nsHttpTransaction::mConnInfo, r=necko-reviewers,valentin
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D243146
# ==============================================================================

diff -r 21cdd98a6828 -r b3ffe3409712 netwerk/protocol/http/TRRServiceChannel.cpp
--- a/netwerk/protocol/http/TRRServiceChannel.cpp	Fri Mar 28 17:48:02 2025 +0000
+++ b/netwerk/protocol/http/TRRServiceChannel.cpp	Fri Mar 28 17:56:38 2025 +0000
@@ -1025,7 +1025,7 @@
     // mTransactionPump doesn't hit OnInputStreamReady and call this until
     // all of the response headers have been acquired, so we can take
     // ownership of them from the transaction.
-    mResponseHead = mTransaction->TakeResponseHead();
+    mResponseHead = mTransaction->TakeResponseHeadAndConnInfo(nullptr);
     if (mResponseHead) {
       uint32_t httpStatus = mResponseHead->Status();
       if (mTransaction->ProxyConnectFailed()) {